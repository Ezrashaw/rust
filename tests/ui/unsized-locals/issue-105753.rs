fn foo() -> [i32] {
    //~^ ERROR the size for values of type `[i32]` cannot be known at compilation time
    todo!()
}

fn main() {
    let x = foo();
    //~^ ERROR the size for values of type `[i32]` cannot be known at compilation time
}
