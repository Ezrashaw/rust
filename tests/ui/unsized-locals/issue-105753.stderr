error[E0277]: the size for values of type `[i32]` cannot be known at compilation time
  --> $DIR/issue-105753.rs:1:13
   |
LL | fn foo() -> [i32] {
   |             ^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `Sized` is not implemented for `[i32]`
   = note: the return type of a function must have a statically known size

error[E0277]: the size for values of type `[i32]` cannot be known at compilation time
  --> $DIR/issue-105753.rs:7:9
   |
LL |     let x = foo();
   |         ^ doesn't have a size known at compile-time
   |
   = help: the trait `Sized` is not implemented for `[i32]`
   = note: all local variables must have a statically known size
   = help: unsized locals are gated as an unstable feature

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
